opus and libopusenc headers are not found on Silicon macOS installed with Homebrew

Homebrew on macOS using Apple Silicon does not install under `/usr/include/opus/`, but under the `/opt/homebrew/include`


```stderr
   Compiling opusenc-sys v0.2.2
error: failed to run custom build command for `opusenc-sys v0.2.2`

Caused by:
  process didn't exit successfully: `/Users/viktorsandstrom/Documents/rust/tau-radio/target/debug/build/opusenc-sys-cbed859fac7156b2/build-script-build` (exit status: 101)
  --- stdout
  cargo:rustc-link-lib=opusenc
  cargo:rerun-if-changed=wrapper.h
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64-apple-darwin
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64_apple_darwin
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-changed=wrapper.h

  --- stderr
  wrapper.h:1:10: fatal error: 'opus/opus_defines.h' file not found

  thread 'main' (9635381) panicked at /Users/viktorsandstrom/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/opusenc-sys-0.2.2/build.rs:13:10:
  Unable to generate bindings: ClangDiagnostic("wrapper.h:1:10: fatal error: 'opus/opus_defines.h' file not found\n")
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

--- a/.github/actions/container/Dockerfile
+++ b/.github/actions/container/Dockerfile
@@ -1,6 +1,6 @@
 FROM rust:slim-bookworm
 
-RUN apt-get update && apt-get install -y libclang-dev libopusenc-dev
+RUN apt-get update && apt-get install -y libclang-dev libopusenc-dev pkg-config
 RUN rustup component add clippy rustfmt
 
 ENTRYPOINT ["sh", "-c"]

--- a/opusenc-sys/Cargo.toml
+++ b/opusenc-sys/Cargo.toml
@@ -13,4 +13,5 @@ links = "opusenc"
 [dependencies]
 
 [build-dependencies]
-bindgen = "0.69.2"
+bindgen = "0.72.1"
+pkg-config = "0.3.32"

--- a/opusenc-sys/build.rs
+++ b/opusenc-sys/build.rs
@@ -4,9 +4,19 @@ fn main() {
     println!("cargo:rustc-link-lib=opusenc");
     println!("cargo:rerun-if-changed=wrapper.h");
 
+    let library = pkg_config::Config::new()
+        .atleast_version("0.2")
+        .probe("libopusenc")
+        .expect("unable to find libopusenc library using pkg-config");
+
     let bindings = bindgen::Builder::default()
         .header("wrapper.h")
-        .clang_args(["-isystem", "/usr/include/opus"])
+        .clang_args(
+            library
+                .include_paths
+                .iter()
+                .map(|path| format!("-I{}", path.to_string_lossy())),
+        )
         .parse_callbacks(Box::new(bindgen::CargoCallbacks::new()))
         .default_enum_style(bindgen::EnumVariation::ModuleConsts)
         .generate()

--- a/opusenc-sys/wrapper.h
+++ b/opusenc-sys/wrapper.h
@@ -1,2 +1,2 @@
-#include <opus/opus_defines.h>
-#include <opus/opusenc.h>
+#include "opus_defines.h"
+#include "opusenc.h"

